build.sc

```diff
@@ -11,7 +11,7 @@ def millBinaryVersion(millVersion: String) = scalaNativeBinaryVersion(millVersio
 object `mill-tpolecat` extends Cross[MillTpolecatCross](millVersions: _*)
 class MillTpolecatCross(millVersion: String) extends ScalaModule with PublishModule {
   override def millSourcePath = super.millSourcePath / os.up
-  override def artifactName   = s"${super.artifactName()}_mill${millBinaryVersion(millVersion)}"
+  override def artifactName   = s"${millModuleSegments.parts.init.mkString("-")}_mill${millBinaryVersion(millVersion)}"
```

Comment:
The Scala binary suffix should most probably not be part of `artifactName`. That's what `artifactId` is for.

Reply:
Do you recommend: ` override def artifactId   = s"${artifactName().split("-").init.mkString("-")}_mill${millBinaryVersion(millVersion)}"` instead?

Reply:
The name adaption can stay in `artifactName`, but I wonder why it was necessary at all, to manually handle the suffix?

These are the definitions in ScalaModule.

```scala
  def artifactSuffix: T[String] = s"_${artifactScalaVersion()}"
  override def artifactId: T[String] = artifactName() + artifactSuffix()
```

And `artifactId` is what is used by the publish module.

Reply:
When we use `Cross` it includes the version as part of the `millModuleSegments.parts`. We need to remove it.

Reply:
Oh I need to use `CrossModuleBase`:
See https://github.com/DavidGregory084/mill-tpolecat/pull/13/files

Reply:
Ok, this is related to the artifactName, which makes sense. But why do you have to add to suffix here? I think, you should end up with two suffixes when publishing, and I wonder why you aren't.

Reply:
Sounds like it's being done the same way in mill-integrationtest: https://github.com/lefou/mill-integrationtest/blob/main/build.sc#L56

Reply:
But with your change in in PR #13, you fixed it.

Reply:
Great, thanks for commenting here! The PR #13 version is way better.

Reply:
I'm sorry for all that fuzz, I misread and it was correct before too. I believed I saw a scala version, but it already only was about the mill version. I'm sorry.